### PR TITLE
[ENG-626] - Feedback

### PR DESCRIPTION
**Pull request by ameer2468**
- Added dialog for feedback
- Landing page has a new endpoint that triggers a slack webhook which sends feedback to a channel in our slack.
- Fixed a bug, added e.stopPropagation to inputs. **The 'Space' shortcut being used in explorer was reacting to existing event bubbling that wasn't being properly prevented, which caused our inputs not to support spacing.**

**Screenshot**:

Screenshot 2023-05-25 at 7 22 45 PM

### CODE REVIEW

File: apps/landing/src/app/api/feedback/route.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+		};
+		await fetch(slackWebhook, {
+			method: 'POST',
+			body: JSON.stringify(slackMessage)
+		});
+		return new Response(undefined, {
+			status: 204
+		});
+	} catch (error) {
+		console.log(error)
```
Comment (Brendonovich) [inline apps/landing/src/app/api/feedback/route.ts:52]: `console.error` here would be nicer

File: apps/landing/src/app/api/feedback/route.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+				message: result.error.toString()
+			}),
+			{
+				status: 400
+			}
+		);
+	}
+	try {
+		const { feedback, emoji } = result.data;
+		const slackWebhook = env.SLACK_FEEDBACK_URL;
```
Comment (Brendonovich) [inline apps/landing/src/app/api/feedback/route.ts:32]: may as well do this inline in the `fetch`

File: interface/app/$libraryId/Layout/Sidebar/FeedbackDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+	emoji: z.string().max(1)
+});
+
+type formType = z.infer<typeof schema>;
+
+export default function FeedbackDialog(props: UseDialogProps) {
+	const form = useZodForm({ schema: schema });
+	const emojis = ['🤩', '😀', '🙁', '😭'];
+	const [emojiSelected, setEmojiSelected] = useState<string | undefined>(undefined);
+	const feedback_url = 'https://spacedrive.com/api/feedback';
```
Comment (Brendonovich) [inline interface/app/$libraryId/Layout/Sidebar/FeedbackDialog.tsx:18]: `schema` doesn't need to be specified twice, `emojis` and `feedback_url` should be stored outside the component (imo in `SCREAMING_SNAKE_CASE` consts)

File: interface/app/$libraryId/Layout/Sidebar/FeedbackDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+	const emojiSelectHandler = (index: number) => {
+		setEmojiSelected(emojis[index]);
+		form.setValue('emoji', emojis[index] as string);
+	};
+
+	const formSubmit = async (data: formType) => {
```
Comment (Brendonovich) [inline interface/app/$libraryId/Layout/Sidebar/FeedbackDialog.tsx:25]: use `form.handleSubmit` here so `formType` isn't necessary

File: packages/ui/src/Dialog.tsx [hunk 5]
```diff
@@ -187,9 +188,10 @@ export function Dialog<S extends FieldValues>({
 								</div>
 								<div className="flex flex-row justify-end space-x-2 border-t border-app-line bg-app-selected p-3">
 									{form.formState.isSubmitting && <Loader />}
-
+									{props.buttonsSideContent && (
+										<div>{props.buttonsSideContent}</div>
+									)}
```
Comment (Brendonovich) [inline packages/ui/src/Dialog.tsx:193]: i hate this but it can stay _for now_

**Review by Brendonovich — CHANGES_REQUESTED** [review]
great work, just got a bunch of minor changes to make things a bit nicer

**Review by Brendonovich — APPROVED** [review]

**Review by utkubakir — APPROVED** [review]

**Pull request merged**